### DISCUSSION TITLE
I can't insert data into the database

### DISCUSSION THREAD

**Discussion by SavixDR** [supabase/supabase#30236] [Questions] [answered]
I have this table called "Stores_Products_Designs" in my project. I am inserting a row into the database. I have disabled RLS for now because the website I'm building doesn't have a Login for now. 

`import { createClient } from "@/utils/supabase/ssr_client/server";
import { NextRequest, NextResponse } from "next/server";

interface ListProps {
	store_code: string;
	sage_code: string;
	design_code: string;
}

const addToList = async ({ store_code, sage_code, design_code }: ListProps) => {
	const supabase = createClient();
	try {
		console.log("Adding to list: ", store_code, sage_code, design_code);
		const { data, error } = await supabase
			.from("Stores-Products-Designs")
			.insert([
				{
					Store_Code: `${store_code}`,
					Product_Sage_Code: `${sage_code}`,
					Design_Code: `${design_code}`,
				},
			])
			.select();
		console.log(data, error);
		return data;
	} catch (e) {
		console.error("Unexpected error:", e);
		throw e;
	}
};

export async function POST(request: NextRequest) {
	try {
		const body = await request.json();
		const {
			store_code: storeCode,
			sage_code: sageCode,
			design_code: designCode,
		} = body;

		// Add to list
		const response = await addToList({
			store_code: storeCode,
			sage_code: sageCode,
			design_code: designCode,
		});

		console.log("Response: ", response);

		if (!response) {
			return NextResponse.json(
				{ error: "Failed to add to list" },
				{ status: 500 }
			);
		}
		return NextResponse.json(
			{ success: true, data: response },
			{ status: 200 }
		);
	} catch (e) {
		console.error("Unexpected error:", e);
		throw e;
	}
}
`
I have double checked the columns names as well. But I can't understand the reason for the error here.  Because rows won't insert into the database. Also I need to know why there will be no error shown. It shown as null

**Comment by GaryAustin1** [comment, marked as answer]
You need to console.log the error as by default Supabase API does not use the try/catch method and just returns the error in error.
You can also look in the Supabase dashboard Gateway API logs or Postgres logs for more error info.

**Reply by SavixDR** [reply]
Thank you very much. I found the error because of this.
